### PR TITLE
Add visibility_mode connectors setting

### PR DESCRIPTION
**Pull request by vallbull**

### CODE REVIEW

File: lib/dl_api_lib/dl_api_lib/connector_availability/base.py [hunk 1]
```diff
@@ -118,13 +121,15 @@ class Connector(ConnectorBase):
     conn_type: ConnectionType = attr.ib(
         converter=lambda v: ConnectionType(v) if not isinstance(v, ConnectionType) else v
     )
-    availability: ConnectorAvailability = attr.ib(default=ConnectorAvailability.free, converter=ConnectorAvailability)
+    visibility_mode: ConnectorAvailability = attr.ib(
+        default=ConnectorAvailability.free, converter=ConnectorAvailability
+    )
```
Comment (KonstantAnxiety) [inline lib/dl_api_lib/dl_api_lib/connector_availability/base.py:126]: Just wondering, why not leave `availability`?

File: lib/dl_api_lib/dl_api_lib/connector_availability/base.py [hunk 2]
```diff
@@ -240,4 +246,4 @@ def check_connector_is_available(self, conn_type: ConnectionType) -> bool:
             LOGGER.warning("Connector %s is not available in current env", conn_type.name)
             return False
 
-        return conn_options.availability == ConnectorAvailability.free
+        return conn_options.visibility_mode == ConnectorAvailability.free
```
Comment (KonstantAnxiety) [inline lib/dl_api_lib/dl_api_lib/connector_availability/base.py:249]:
```suggestion
        return conn_options.visibility_mode != ConnectorAvailability.uncreatable
```

**Review by KonstantAnxiety — CHANGES_REQUESTED** [review]

File: lib/dl_api_lib/dl_api_lib/connector_availability/base.py [hunk 3]
```diff
@@ -76,6 +76,8 @@ def as_dict(self, localizer: Localizer) -> dict[str, Any]:
 
 @attr.s(kw_only=True)
 class ConnectorBase(LocalizedSerializable, metaclass=abc.ABCMeta):
+    visibility_mode: ConnectorAvailability = None
```
Comment (KonstantAnxiety) [inline lib/dl_api_lib/dl_api_lib/connector_availability/base.py:79]: For the sake of consistency, let's make is a prop, just like `hidden` or `conn_type_str`

**Review by KonstantAnxiety — CHANGES_REQUESTED** [review]

File: lib/dl_api_lib/dl_api_lib/connector_availability/base.py (hunk 1 above)
Comment (vallbull) [reply lib/dl_api_lib/dl_api_lib/connector_availability/base.py:126]: When we discussed it with Evgenyi, we agreed on such a name and I stuck to it

**Review by KonstantAnxiety — APPROVED** [review]

**Pull request merged**